ctapipe/io/simteleventsource.py

```diff
@@ -471,13 +476,17 @@ def _fill_trigger_info(data, array_event):
             data.trigger.event_type = EventType.UNKNOWN
 
         data.trigger.tels_with_trigger = trigger["triggered_telescopes"]
+        if self.allowed_tels:
+            data.trigger.tels_with_trigger = np.intersect1d(
+                data.trigger.tels_with_trigger, np.array(list(self.allowed_tels))
```

Comment:
calling `array(list(...` every time is not necessary.

Reply:
Should I call it once in the init?

Reply:
This should just be `subarray.tel_ids`, which is cached.

Reply:
Also, https://numpy.org/doc/stable/reference/generated/numpy.intersect1d.html

We can `assume_unique` here, right? No duplicated tel ids